bodhi/server/models.py

```diff
@@ -1732,7 +1732,7 @@ def set_request(self, db, action, username):
         # If status is testing going to stable request and action is revoke,
         # keep the status at testing
         elif self.status is UpdateStatus.testing and self.request is UpdateRequest.stable \
-                and action is UpdateRequest.revoke:
+                or UpdateRequest.batched and action is UpdateRequest.revoke:
```

Comment:
You could also replace ```(self.request is UpdateRequest.stable or self.request is UpdateRequest.batched)``` in my suggestion with ```self.request in (UpdateRequest.stable, UpdateRequest.batched)```, eliminating the ```or``` and maybe a little shorter too.